tests/numerics/smoothness_estimator_02.output

```diff
@@ -0,0 +1,22 @@
+DEAL::-----------------------------------
+DEAL::1d, p=1, max_p=4, N=4
+DEAL::-----------------------------------
+DEAL::Linf norm in exact and calculate Fourier coefficients:
+DEAL::0.0161235
+DEAL::sobolev index:
+DEAL::0.797851
```

Comment:
If you have a look at the 1D case for the linear function (p=1), you'll notice that the coefficients decay with `k^(-1)` if you omit the zeroth coefficient. Thus, the linear regression yields the correct slope of `1`, since we later subtract `1/2` and return a Sobolev index of about `1/2`. The slope is actually a bit larger because of the zeroth coefficient.

However, I'm not sure if a Sobolev index of `1/2` is correct for the linear test function in 1D. I would assume that it should be `1` at least. Is the subtraction of `dim/2` really necessary?

Reply:
@bangerth and I talked about this during the last workshop: Since we determine the Fourier expansion on each cell's domain, we have to consider the expanded solution on the global domain as a periodic continuation of the former. This may lead to steep gradients or even discontinuities on connecting faces. This circumstance should also have an effect on the Fourier coefficients, and thus ultimately on their decay. So we really need to treat these values as smoothness indicators, rather than actual Sobolev indices.

Other strategies exist to directly approximate Sobolev indices, see [[Ainsworth1998, sec. 2.4]](https://dx.doi.org/10.1016/S0045-7825(97)00101-1). We could do this by solving the equation on smaller patches and verify how the error decays, but this tends to be expensive. Thus we won't pay further attention to this topic (for now).